### ISSUE TITLE
[IMPROVE] UI Toolkit Support

### ISSUE THREAD

**Issue by Wuzseen** [mackysoft/Unity-SerializeReferenceExtensions#51]
### Improvement destription

Hello! This is a great little extension--makes SerializeReference much more useful out of the box.

Due to Unity's restrictions/issues with UI tooolkit and IMGUI inspectors clashing, it would be nice to have UI Toolkit version of the subclass selector implemented.

I did this locally just to quickly unblock something. Simply added this method to SubclassSelectorDrawer. I'd make a PR but I'm not sure this is how you'd want to do it with your package anyway. It doesn't totally match the styling of your IMGUI implementation.

`
        public override VisualElement CreatePropertyGUI(SerializedProperty property)
        {
            VisualElement root = new();

            if (property.propertyType == SerializedPropertyType.ManagedReference)
            {
                Button dropdownButton = null;
                dropdownButton = new Button(() =>
                {
                    TypePopupCache popup = this.GetTypePopup(property);
                    this.m_TargetProperty = property;
                    popup.TypePopup.Show(dropdownButton.contentRect);
                });

                dropdownButton.text = this.GetTypeName(property).text;
                root.Add(dropdownButton);

                PropertyField propertyField = new(property);
                root.Add(propertyField);
            }
            else
            {
                TextField text = new(SubclassSelectorDrawer.k_IsNotManagedReferenceLabel.text);
                root.Add(text);
            }

            return root;
        }
`

**Comment by mackysoft** [comment]
@Wuzseen 
Sorry for the delay.

To apply an implementation using VisualElements to this library, it is necessary to resolve issues related to version compatibility and drawing compatibility with IMGUI. Therefore, the priority of implementation in my roadmap is likely to be low.

Please feel free to reopen if there is anything else. Contributions are welcome!

**Comment by mackysoft** [comment]
A little report on some work. I've been working on the UIToolkit for the past few days, but with the current Unity Editor API, it was difficult to completely replace the current IMGUI implementation.

The inability to apply the SerializedProperty context menu to custom field created with UIElements is one of the main reasons.
https://discussions.unity.com/t/any-plans-for-a-begin-endproperty-equivalent/892424
